websocket/src/websocket.cpp

```diff
@@ -346,9 +351,17 @@ static int LuaSend(lua_State* L)
 
     wslay_event_queue_msg(conn->m_Ctx, &msg); // it makes a copy of the data
 #else
-
-    dmSocket::Result sr = Send(conn, string, string_length, 0);
-    if (dmSocket::RESULT_OK != sr)
+    EMSCRIPTEN_RESULT result;
+    int write_mode = dmScript::CheckTableNumber(L, 3, "type", DATA_TYPE_BINARY);
+    if (write_mode == DATA_TYPE_BINARY)
+    {
+        result = emscripten_websocket_send_binary(conn->m_WS, (void*)string, string_length);
+    }
+    else
+    {
+        result = emscripten_websocket_send_utf8_text(conn->m_WS, string);
+    }
+    if (result)
```

Comment:
This stuff is why I added socket.cpp, to act as a wrapper, to make this sort of code more readable. Is this not possible here?